src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.TaskList;
+import tasks.ToDo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+
+public class Storage {
+
+    public String filePath;
+
+    public Storage (String filePath) {
+        this.filePath = filePath;
+    }
+
+    public void saveData() throws IOException {
+        try {
+            File file = new File(this.filePath);
+
+            boolean createDirectory = file.getParentFile().mkdirs();
+            boolean created = file.createNewFile();
+            FileWriter fileWriter = new FileWriter(file, false);
+            for (Task task : TaskList.dukeList) {
+                String text = "";
+                String description = task.getDescription();
+                //task.isDone = false;
+                if (task instanceof ToDo) {
+                    text = "T | " + (task.getIsDone() ? "1" : "0") + " | " + description + '\n';
+                } else if (task instanceof Event) {
+                    Event event = ((Event) task);
+                    text = "E | " + (task.getIsDone() ? "1" : "0") + " | " + task.getDescription().trim() + " | " + event.getTime() + '\n';
+                } else {
+                    Deadline deadline = ((Deadline) task);
+                    text = "D | " + (task.getIsDone() ? "1" : "0") + " | " + task.getDescription().trim() + " | " + deadline.getTime() + '\n';
+                }
+                fileWriter.write(text);
+                fileWriter.flush();
+            }
+        } catch(IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    public void loadData() throws IOException {
+        File file = new File(this.filePath);
+        boolean createDirectory = file.getParentFile().mkdirs();
+        boolean created = file.createNewFile();
+
+        Scanner sc = new Scanner(file);
+        while (sc.hasNextLine()) {
+            String input = sc.nextLine();
+            String[] commandWords = input.split(" \\| ", 4);
+            Task task = null;
+            switch (commandWords[0]) {
+                case "D":
```

Comment:
Note the indentation for switch case, switch and case should be on the same indentation level